libs/ngx-dsfr/checkbox/src/lib/checkbox.component.ts

```diff
@@ -0,0 +1,157 @@
+  'You MUST provide a value for the items attribute 😡 !!!';
+export const EMPTY_FAILURE_MESSAGE_ERROR: string =
+  'You MUST provide a value for the failureMessage attribute when hasFailed is true 😡 !!!';
+export const EMPTY_SUCCESS_MESSAGE_ERROR: string =
+  'You MUST provide a value for the successMessage attribute when hasSucceeded is true 😡 !!!';
+
+export interface CheckboxItem {
+  id: string;
+  label: string;
+  value: string;
+  hint?: string;
+}
+
+@Component({
+  selector: 'dsfr-checkbox',
+  templateUrl: './checkbox.component.html',
+  styleUrls: ['./checkbox.component.scss'],
+  providers: [
+    {
+      provide: NG_VALUE_ACCESSOR,
+      useExisting: forwardRef(() => DsfrCheckboxComponent),
+      multi: true
+    }
+  ]
+})
+
+export class DsfrCheckboxComponent implements ControlValueAccessor, OnInit, OnChanges {
+  @Input() legend = '';
+  @Input() name = '';
+  @Input() fieldSetId = '';
+  @Input() hint = '';
+  @Input() items: CheckboxItem[] = [];
+  @Input() inline = false;
+  @Input() disabled = false;
+  @Input() size = ElementSize.MEDIUM;
+  @Input() hasFailed = false;
+  @Input() failureMessage = '';
+  @Input() hasSucceeded = false;
+  @Input() successMessage = '';
+
+  onChange = (_: string[]) => { };
+  onTouched = (_: string[]) => { };
+
+  fieldSetClasses: Record<string, boolean> = {};
+  ariaLabelledBy: string | null = null;
+
+  get values(): string[] {
+    return this._values;
+  }
+
+  set values(val: string[]) {
+    this.writeValue(val);
+  }
+
+  private _values!: string[];
+
+  ngOnChanges(changes: SimpleChanges): void {
+    if (this.hasFailed && !this.failureMessage) {
+      throw EMPTY_FAILURE_MESSAGE_ERROR;
+    }
+    if (this.hasSucceeded && !this.successMessage) {
+      throw EMPTY_SUCCESS_MESSAGE_ERROR;
+    }
+    this.setFieldSetClasses();
+    this.setAriaLabelledBy();
+  }
+
+  ngOnInit(): void {
+    if (!this.legend) {
+      throw EMPTY_LEGEND_ERROR;
+    }
+    if (!this.name) {
+      throw EMPTY_NAME_ERROR;
+    }
+    if (!this.fieldSetId) {
+      throw EMPTY_FIELDSET_ID_ERROR;
+    }
+    if (this.items.length === 0) {
+      throw EMPTY_ITEMS_ERROR;
+    }
+  }
+
+  private setFieldSetClasses() {
+    this.fieldSetClasses = {
+      'fr-fieldset--inline': this.inline,
+      'fr-fieldset--error': this.hasFailed,
+      'fr-fieldset--valid': this.hasSucceeded
+    };
+  }
+
+  private setAriaLabelledBy() {
+    if (this.hasFailed) {
+      this.ariaLabelledBy = `${this.fieldSetId} ${this.fieldSetId}-error`;
+      return;
+    }
+    if (this.hasSucceeded) {
+      this.ariaLabelledBy = `${this.fieldSetId} ${this.fieldSetId}-valid`;
+      return;
+    }
+    this.ariaLabelledBy = this.fieldSetId;
+  }
+
+  registerOnChange(fn: (_: string[]) => void): void {
+    this.onChange = fn;
+  }
+
+  registerOnTouched(fn: (_: string[]) => void): void {
+    this.onTouched = fn;
+  }
+
+  writeValue(value: string[]): void {
+    this._values = value;
+    this.onChange(value);
```

Comment:
OK... so... I was wondering why you would do this in `writeValue`, then I noticed that you started from the code for `RadioComponent`...
Which means that, ultimately, this is also my fault, since I let it go and authorized it when I made the Code Review 🤦🏻‍♂️

Anyway...

Could you try without the call to `onChange` please ? 😇

It's supposed to be called when the user has changed the value in the UI , so to communicate information in that direction : 
```
<Component with ControlValueAccessor> => <FormControl associated to it in the parent Component>
```

Whereas, `writeValue` is called by the Angular Form API automatically, when the value changes programmatically from the outside, so it goes in the other direction : 
```
<FormControl associated to it in the parent Component> => <Component with ControlValueAccessor>
```